packages/vite/src/node/config.ts

```diff
@@ -469,6 +478,8 @@ export async function resolveConfig(
 
   const optimizeDeps = config.optimizeDeps || {}
 
+  config.isSPA ??= true
```

Comment:
I was thinking that you'd have something like `isSPA: true,` below in the creation of `resolved` and then you'd use `resolved` when checking the value of `isSPA`

Reply:
Actually, I've just pushed a commit to resolve `isSPA`'s default value at the very end, just before it's used.

This provides more flexibility: it allows the framework to overwrite Vite's default value while enabling the user to overwrite the framework provided default value.

Let me elaborate.

For example, this is how vite-plugin-ssr and Telefunc are ensuring that the server and preview default port is `3000` for Vite 3 (despites the fact that Vite 3 changed the default port from `3000` to `5173` which I don't like as I find it difficult to remember):

https://github.com/vikejs/telefunc/blob/62d2753fe452e3bb5981da110b7e1170b35b1c61/telefunc/node/vite/plugins/previewConfig.ts#L23-L27

The neat thing here is that user can still set a different port.

That works because Vite resolves the default value of `config.server.port` and `config.preview.port` at the very end:

https://github.com/vitejs/vite/blob/e37a51dc767fe3fcf2eb49f7f76eab63d90f777a/packages/vite/src/node/server/index.ts#L572-L580
https://github.com/vitejs/vite/blob/e37a51dc767fe3fcf2eb49f7f76eab63d90f777a/packages/vite/src/node/preview.ts#L116-L122

I'm marking this conversation as resolved but let me know if you disagree and I'll unresolve it.